nixos/modules/services/web-apps/freshrss.nix

```diff
@@ -0,0 +1,247 @@
+      enable = true;
+      virtualHosts.${cfg.virtualHost} = {
+        root = "${cfg.package}/p";
+
+        locations."~ ^.+?\.php(/.*)?$".extraConfig = ''
+          fastcgi_pass unix:${config.services.phpfpm.pools.${cfg.pool}.socket};
+          fastcgi_split_path_info ^(.+\.php)(/.*)$;
+          include ${pkgs.nginx}/conf/fastcgi_params;
+          include ${pkgs.nginx}/conf/fastcgi.conf;
+        '';
+
+        locations."/" = {
+          tryFiles = "$uri $uri/ index.php";
+          index = "index.php index.html index.htm";
+        };
+      };
+    };
+
+    # Set up phpfpm pool
+    services.phpfpm.pools = mkIf (cfg.pool == poolName) {
+      ${poolName} = {
+        user = "freshrss";
+        settings = {
+          "listen.owner" = "nginx";
+          "listen.group" = "nginx";
+          "listen.mode" = "0600";
+          "pm" = "dynamic";
+          "pm.max_children" = 32;
+          "pm.max_requests" = 500;
+          "pm.start_servers" = 2;
+          "pm.min_spare_servers" = 2;
+          "pm.max_spare_servers" = 5;
+          "catch_workers_output" = true;
+          "php_admin_value[open_basedir]" = "${cfg.package}:${cfg.dataDir}:/tmp";
+        };
+        phpEnv = {
+          FRESHRSS_DATA_PATH = "${cfg.dataDir}";
+        };
+      };
+    };
+
+    users.users.freshrss = {
+      description = "FreshRSS service user";
+      isSystemUser = true;
+      group = "freshrss";
+    };
+    users.groups.freshrss = {};
+
+    systemd.services.freshrss-config = let
+      settingsFlags = concatStringsSep " \\\n    "
+        (mapAttrsToList (k: v: "${k} ${toString v}") {
+        "--default_user" = ''"${cfg.defaultUser}"'';
+        "--auth_type" = ''"form"'';
+        "--base_url" = ''"${cfg.baseUrl}"'';
+        "--language" = ''"${cfg.language}"'';
+        "--db-type" = ''"${cfg.database.type}"'';
+        # The following attributes are optional depending on the type of
+        # database.  Those that evaluate to null on the left hand side
+        # will be omitted.
+        ${if cfg.database.name != null then "--db-base" else null} = ''"${cfg.database.name}"'';
+        ${if cfg.database.passFile != null then "--db-password" else null} = ''"$(cat ${cfg.database.passFile})"'';
+        ${if cfg.database.user != null then "--db-user" else null} = ''"${cfg.database.user}"'';
+        ${if cfg.database.tableprefix != null then "--db-prefix" else null} = ''"${cfg.database.tableprefix}"'';
+        ${if cfg.database.host != null && cfg.database.port != null then "--db-host" else null} = ''"${cfg.database.host}:${toString cfg.database.port}"'';
+      });
+    in {
+      description = "Set up the state directory for FreshRSS before use";
+      wantedBy = [ "multi-user.target" ];
+      serviceConfig = {
+        Type = "oneshot";
+        User = "freshrss";
+        Group = "freshrss";
+        StateDirectory = "freshrss";
+        WorkingDirectory = cfg.package;
+      };
```

Comment:
Could you also add more hardening options to this and to the other service as well.

Reply:
Something like [this](https://github.com/NixOS/nixpkgs/blob/c23131bc34da36f938c6f64de685afbd0f149039/nixos/modules/services/security/endlessh-go.nix#L98-L129). Just run your service through `systemd-analyze security` and patch out some holes.

Reply:
is this fine 24a1171cf5c7762176f599aa9b87b87169906a11

Reply:
Yep, looks much better now.